docs/design_docs/index_design.md

```diff
@@ -0,0 +1,191 @@
+to filter out the tasks that need to be allocated, including unallocated tasks and tasks that have been suspended by
+the corresponding IndexNode. Then sort according to the version size of each task, and assign tasks with a smaller 
+version first. The purpose is to prevent certain special tasks from occupying resources all the time and always fail
+to execute successfully. When a task is assigned, its corresponding Version is increased by one. Then send the task to
+IndexNode for execution, and update the index status in the MetaTable.
+
+### 8.3.7 recycleUnusedIndexFiles
+
+Delete useless index files, including lower version index files and index files corresponding to the deleted index. 
+In order to distinguish whether the low version index file corresponding to the index has been cleaned up, recycled is
+introduced as a mark. Only after the index task is completed will the lower version index files be cleaned up, and the 
+index file corresponding to the lower version index file that has been cleaned up is marked as True.
+
+This is also a timer, which periodically traverses the MetaTable to obtain the index corresponding to the index file
+that needs to be cleaned up. If the index is marked as deleted, the information corresponding to the index is deleted
+in the MetaTable. Otherwise, it just cleans up the index file of the lower version.
+
+## 8.4 IndexNode Create Index
+
+IndexNode is the execution node of index building tasks, and all index building tasks are forwarded to IndexNode by
+IndexCoordinate for execution. When IndexNode executes an index build request, it first reads IndexMeta information
+from ETCD, and checks whether the index task is marked for deletion when IndexCoordinate is forwarded to IndexNode.
+If it is marked as deleted, then there is no need to actually build the index, just mark the index task status as 
+completed, and then write it to ETCD. When IndexCoordinate perceives that the status corresponding to the index is 
+complete, it deletes the index task from the MetaTable. If it is checked that the index is not marked for deletion,
+then the index needs to be built. The original data must be loaded first when building the index. The original data
+is stored in minIO/S3, and the storage path is notified by RootCoordinate in the index build request. After loading the
+original data, the data is deserialized into data blocks, and then cgo is called to build the index. When the index is
+built, the index data is serialized into data blocks, and then written into the file. The directory organization of the
+index file is "indexBuildID" /IndexTaskVersion/partitionID/segmentID/key", where key corresponds to the serialized key
+of index data. After the index is built, record the index file directory in IndexMeta, and then write it to ETCD.
+
+## 8.5 API
+
+### 8.5.1 BuildIndex
+
+Index building is asynchronous, so when an index building request comes, an IndexBuildID is assigned to the task and 
```

Comment:
we need a diagram show the state machine of index building task 
for example:
Start -> process -> failure/success

Reply:
IndexState is the state I mean